api/v1alpha1/managedocs_types.go

```diff
@@ -38,9 +38,27 @@ type ManagedOCSSpec struct {
 	ReconcileStrategy ReconcileStrategy `json:"reconcileStrategy,omitempty"`
 }
 
+type ComponentStatus string
+
+const (
+	ComponentReady   ComponentStatus = "Ready"
+	ComponentPending ComponentStatus = "Pending"
+	ComponentNone    ComponentStatus = "None"
+)
+
+type ManagedOCSComponent struct {
+	Status      ComponentStatus `json:"status"`
+	LastChecked metav1.Time     `json:"lastChecked"`
```

Comment:
`LastCheck` is a misleading name for the field because for some resources we can be much more accurate by reading a last update time field on the resource and use it here instead of taking the last check date.

My suggestion is to rename to `LastUpdate` or `LastUpdateTime`